src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java

```diff
@@ -195,32 +199,26 @@ public SubmissionApiDelegateImpl(
 
     }
 
-    private UserDetails buildUserDetails(Submission submission) {
+    private UserDetails buildUserDetails() {
 
         UserDetails userDetails = new UserDetails();
 
-        if(submission.getAccountDetails() != null) {
+        AccountDetails accountDetails = accountService.getCsoAccountDetails(getUniversalIdFromContext().get());
 
-            if(submission.getAccountDetails().isFileRolePresent()) {
+        if(accountDetails != null) {
+
+            if(accountDetails.isFileRolePresent()) {
                 Account account = new Account();
                 account.setType(Account.TypeEnum.CSO);
-                account.setIdentifier(submission.getAccountDetails().getAccountId().toString());
+                account.setIdentifier(accountDetails.getAccountId().toString());
                 userDetails.addAccountsItem(account);
             }
 
-            userDetails.setUniversalId(submission.getAccountDetails().getUniversalId());
-            userDetails.setFirstName(submission.getAccountDetails().getFirstName());
-            userDetails.setLastName(submission.getAccountDetails().getLastName());
-            userDetails.setMiddleName(submission.getAccountDetails().getMiddleName());
-            userDetails.setEmail(submission.getAccountDetails().getEmail());
-
-        } else {
-
-            // TODO: remove this
-            userDetails.setFirstName("firstName");
-            userDetails.setLastName("lastName");
-            userDetails.setMiddleName("middleName");
-            userDetails.setEmail("email");
+            userDetails.setUniversalId(accountDetails.getUniversalId());
+            userDetails.setFirstName(accountDetails.getFirstName());
+            userDetails.setLastName(accountDetails.getLastName());
+            userDetails.setMiddleName(accountDetails.getMiddleName());
+            userDetails.setEmail(accountDetails.getEmail());
```

Comment:
what is this for? I dont think the userdetails object ever has to have any of these fields populated :)

Reply:
that's good comment but it is legacy and I will remove that in an other PR